daemon/api_cohort.go

```diff
@@ -0,0 +1,59 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/overlord/auth"
+)
+
+var cohortsCmd = &Command{
+	Path: "/v2/cohorts",
+	POST: postCohorts,
+}
+
+func postCohorts(c *Command, r *http.Request, user *auth.UserState) Response {
+	var inst client.CohortAction
+	dec := json.NewDecoder(r.Body)
+	if err := dec.Decode(&inst); err != nil {
+		return BadRequest("cannot decode request body into cohort instruction: %v", err)
+	}
+	if dec.More() {
+		return BadRequest("spurious content after cohort instruction")
+	}
+
+	if inst.Action != "create" {
+		return BadRequest("unknown cohort action %q", inst.Action)
+	}
+
+	if len(inst.Snaps) == 0 {
+		return BadRequest("snaps cannot be empty")
```

Comment:
```suggestion
		return BadRequest("cohort must relate to at least one snap")
```

Reply:
yes i agree the error message is poor.
but, there is no 'relation': you're asking for a cohort key for each of those snaps, not one cohort that covers them all.
`"cannot request cohorts for 0 snaps"`?

Reply:
or, just detect it, accept it, and return a `{}`

Reply:
In the end I'm just returning an empty map in this case